__MACOSX/php-practice/php-practice_2.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+// Q1
+echo "1から100までのカウントを開始します\n";
+
+for( $i = 1; $i <= 100; $i++){
+    if($i % 4 === 0 && $i % 5 === 0){
+        echo "tic-tac\n";
+    }elseif($i % 4 ===0){
+        echo "tic\n";
+    }elseif($i % 5 === 0){
+        echo "tac\n";
+    }else{
+        echo $i."\n";
+    }
+}
+
+// Q2
+$personalInfos = [
+    [
+        'name' => 'Aさん',
+        'mail' => 'aaa@mail.com',
+        'tel'  => '09011112222'
+    ],
+    [
+        'name' => 'Bさん',
+        'mail' => 'bbb@mail.com',
+        'tel'  => '08033334444'
+    ],
+    [
+        'name' => 'Cさん',
+        'mail' => 'ccc@mail.com',
+        'tel'  => '09055556666'
+    ],
+];
+
+// 問題1
+$searchTel = 'Bさん';
+
+// nameから対象のインデックス番号を取得;
+$key = array_search($searchTel,array_column($personalInfos,'name'));
+echo $searchTel."の電話番号は".$personalInfos[$key]['tel']."です。\n";
+
+// 問題2
+foreach ($personalInfos as $x => $y){
+    
+    $no = $x +1;
+    echo $no."番目の".$personalInfos[$x]['name']."のメールアドレスは".$personalInfos[$x]['mail']."で、電話番号は".$personalInfos[$x]['tel']."です。\n";
+}
+
+// 問題3
+$ageList = [25, 30, 18];
+
+foreach ($personalInfos as $c => $d){
+    $personalInfos[$c]['age'] = $ageList[$c];
+}
+var_dump($personalInfos);
+
+// Q3
+class Student
+{
+    public $studentId;
+    public $studentName;
+
+    public function __construct($id, $name)
+    {
+        $this->studentId = $id;
+        $this->studentName = $name;
+    }
+
+    public function attend()
+    {
+        echo '授業に出席しました。';
+    }
+}
+
+$tanaka = new Student('10','越前');
+echo "学籍番号".$tanaka -> studentId."番の生徒は".$tanaka -> studentName."です。";
+
+//Q4
+class Student
+{
+    public $studentId;
+    public $studentName;
+
+    public function __construct($id, $name)
+    {
+        $this->studentId = $id;
+        $this->studentName = $name;
+    }
+
+    public function attend($lesson)
+    {
+        echo $this->studentName.'は'.$lesson.'の授業に出席しました。学籍番号：'.$this->studentId;
+    }
+}
+
+$yamada = new Student(120, '山田');
+$yamada->attend('PHP');
+
+// Q5
+// 問題１
+$now = new DateTime();
+echo $now->format('Y-m-d'."\n");
+
+// 問題3
+$now = new DateTime();
+$date = new DateTime('1992-4-25');
+
+$diff = $date->diff($now);
+echo $diff->format('あの日から%a日経過しました。');
```

Comment:
練習問題2 Q5-2

こちらの2行は、1行で書くことができます！

```suggestion
echo $date->diff($now)->format('あの日から%a日経過しました。');
```

このようにメソッドの後にメソッドを繋げて書く方法を「メソッドチェーン」と言います！

`オブジェクト名->メソッド1()->メソッド2();`
この場合メソッド2は、メソッド1の返り値から呼ばれてます！